components/PaymentForm.js

```diff
@@ -20,9 +19,11 @@ function PaymentForm({ onClose = () => null }) {
   const [url, setURL] = useState('');
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [isModalOpen, toggleModal] = useState(false);
+  const [finalAmount, updateFinalAmount] = useState(getFinalAmount(form.amount));
```

Comment:
No need to use `useState` again here. We already have updated `amount` in `form`. You can just use `form.amount` to calculate `finalAmount`. See this example if its unclear https://codesandbox.io/embed/nifty-visvesvaraya-v2zwu